applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/eigenfrequency_response_function_utility.h

```diff
@@ -64,327 +64,351 @@ namespace Kratos
-
-				perturbed_LHS -= perturbed_mass_matrix;
-
-				aux = prod(perturbed_LHS , eigenvector_of_element);
-				gradient_contribution[0] = inner_prod(eigenvector_of_element , aux);
-
-				node_i.X0() -= mDelta;
-				// End derivative of response w.r.t. x-coord --------------------
-
-				// Reset pertubed RHS and mass matrix
-				perturbed_LHS = Matrix(0,0);
-				perturbed_mass_matrix = Matrix(0,0);
-				aux = Vector(0);
-
-				// Derivative of response w.r.t. y-coord ------------------------
-				node_i.Y0() += mDelta;
-
-				elem_i.CalculateMassMatrix(perturbed_mass_matrix, CurrentProcessInfo);
-
-				perturbed_mass_matrix = (perturbed_mass_matrix - mass_matrix_org) / mDelta;
-				perturbed_mass_matrix *= eigenvalue;
-
-				elem_i.CalculateLocalSystem(perturbed_LHS, dummy ,CurrentProcessInfo);
-
-				perturbed_LHS = (perturbed_LHS - LHS_org) / mDelta;
-
-				perturbed_LHS -= perturbed_mass_matrix;
-
-				aux = prod(perturbed_LHS , eigenvector_of_element);
-				gradient_contribution[1] = inner_prod(eigenvector_of_element , aux);
-				node_i.Y0() -= mDelta;
-				// End derivative of response w.r.t. y-coord --------------------
-
-				// Reset pertubed RHS and mass matrix
-				perturbed_LHS = Matrix(0,0);
-				perturbed_mass_matrix= Matrix(0,0);
-				aux = Vector(0);
-
-				// Derivative of response w.r.t. z-coord ------------------------
-				node_i.Z0() += mDelta;
-
-				elem_i.CalculateMassMatrix(perturbed_mass_matrix, CurrentProcessInfo);
-				perturbed_mass_matrix = (perturbed_mass_matrix - mass_matrix_org) / mDelta;
-				perturbed_mass_matrix *= eigenvalue;
-
-				elem_i.CalculateLocalSystem(perturbed_LHS, dummy ,CurrentProcessInfo);
-				perturbed_LHS = (perturbed_LHS - LHS_org) / mDelta;
-
-				perturbed_LHS -= perturbed_mass_matrix;
-
-				aux = prod(perturbed_LHS , eigenvector_of_element);
-				gradient_contribution[2] =  inner_prod(eigenvector_of_element , aux);
-				node_i.Z0() -= mDelta;
-				// End derivative of response w.r.t. z-coord --------------------
-
-				// Assemble sensitivity to node
-				noalias(node_i.FastGetSolutionStepValue(SHAPE_SENSITIVITY)) += gradient_contribution;
-			}
-		}
-		KRATOS_CATCH("");
-	}
-
-	// ==============================================================================
-
-	///@}
-	///@name Protected  Access
-	///@{
-
-	///@}
-	///@name Protected Inquiry
-	///@{
-
-	///@}
-	///@name Protected LifeCycle
-	///@{
-
-	///@}
+    ///@name Protected static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    // ==============================================================================
+    void CheckSettingsForGradientAnalysis(Parameters& rResponseSettings)
+    {
+        const std::string gradient_mode = rResponseSettings["gradient_mode"].GetString();
+
+        if (gradient_mode.compare("semi_analytic") == 0)
```

Comment:
I don't fully understand what you do here, is this `if(gradient_mode == "semi_analytical")`?

Reply:
https://stackoverflow.com/questions/9158894/differences-between-c-string-and-compare